crates/proc_macro_srv/src/rustc_server.rs

```diff
@@ -287,7 +323,8 @@ impl server::TokenStream for Rustc {
         stream.is_empty()
     }
     fn from_str(&mut self, src: &str) -> Self::TokenStream {
-        Self::TokenStream::from_str(src).expect("cannot parse string")
+        let s = Self::TokenStream::from_str(src).expect("cannot parse string");
+        TokenStream::with_subtree(subtree_replace_token_ids_with_unspecified(s.subtree))
```

Comment:
Maybe it should call in https://github.com/rust-analyzer/rust-analyzer/blob/80ef3f9aca81bf38a753cde3c8c7875af95a966d/crates/proc_macro_srv/src/rustc_server.rs#L186 directly, since we discard the `TokenMap` in there ?